securedrop/tests/functional/test_journalist.py

```diff
@@ -100,8 +100,12 @@ def missing_msg_file(self):
         self._source_logs_out()
 
         # Remove the message file from the store
+        filesystem_id = self.source_app.crypto_util.hash_codename(self.source_name)
         storage_path = Path(self.journalist_app.storage.storage_path)
```

Comment:
If you narrow the search by including the filesystem id in the path, and changing the glob pattern to only find messages, it'll be a little faster and more resistant to other changes that might create files or replies in the directory: 
```python
-        storage_path = Path(self.journalist_app.storage.storage_path)
+        storage_path = Path(self.journalist_app.storage.storage_path) / filesystem_id
         msg_files = [
-            p for p in storage_path.rglob("*")
-            if p.is_file() and filesystem_id in p.parts
+            p for p in storage_path.glob("*-msg.gpg")
```

Reply:
This makes total sense. I will implement the change as suggested.

Reply:
Thanks!